lib/gui/app/components/source-selector/source-selector.tsx

```diff
@@ -335,7 +336,17 @@ export class SourceSelector extends React.Component<
 				path: selected,
 			});
 		}
-		return new sourceDestination.Http({ url: selected });
+		let authorization: string | undefined;
+		if (selected.includes('authorization')) {
```

Comment:
Perhaps this should be `'authorization='` to match the line below? I guess you wouldn't want this to trigger if the image URL was something like `http://www.no-authorization-needed.com/download/os.img` :wink: 

(I guess for completeness, you could do full query-string parsing, in case there are other query-params after the authorization param, but perhaps that's overkill?)